ui/search/nuxeo-search-results-layout.js

```diff
@@ -0,0 +1,149 @@
+import '../nuxeo-layout.js';
+import { I18nBehavior } from '../nuxeo-i18n-behavior.js';
+
+{
+  /**
+   * `nuxeo-search-results-layout`
+   * @group Nuxeo UI
+   * @appliesMixin Nuxeo.I18nBehavior
+   * @element nuxeo-search-results-layout
+   */
+  class SearchResultsLayout extends mixinBehaviors([I18nBehavior], Nuxeo.Element) {
+    static get template() {
+      return html`
+        <nuxeo-layout
+          id="results"
+          href="[[_resultsHref(searchName, hrefBase)]]"
+          model="[[_resultsModel(searchName,nxProvider)]]"
+          error="[[i18n('searchResults.layoutNotFound', searchName)]]"
+          on-element-changed="_formChanged"
+        ></nuxeo-layout>
+      `;
+    }
+
+    static get is() {
+      return 'nuxeo-search-results-layout';
+    }
+
+    static get importMeta() {
+      const importMeta = import.meta;
+      importMeta.url = importMeta.url.replace('/er/ui/', '/elements/');
+      return importMeta;
+      // return import.meta;
+    }
+
+    static get properties() {
+      return {
+        /**
+         * The name of the search layout.
+         */
+        searchName: String,
+        /**
+         * The `nuxeo-page-provider` instance used to perform the search.
+         */
+        nxProvider: HTMLElement,
+        /**
+         * An object propagating key/values served by enclosing slot contents.
+         */
+        model: {
+          type: Object,
+          value() {
+            return {};
+          },
+        },
+        /**
+         * The `nuxeo-results` element bound to this element.
+         */
+        results: {
+          type: Object,
+          notify: true,
+        },
+        /**
+         * The base url for the layout href.
+         */
+        hrefBase: String,
+      };
+    }
+
+    get element() {
+      return this.$.results && this.$.results.element;
+    }
+
+    fetch() {
+      if (this.results) {
+        this.results.fetch();
+      } else {
+        return Promise.resolve();
+      }
+    }
+
+    reset() {
+      if (this.results) {
+        this.results.reset();
+      }
+    }
+
+    _resultsHref(searchName, hrefBase) {
+      const name = searchName.toLowerCase();
+      const base = hrefBase || pathFromUrl((this.__dataHost && this.__dataHost.importPath) || this.importPath);
+      return `${base}${name}/${['nuxeo', name, 'search-results'].join('-')}.html`;
+    }
+
+    _resultsModel() {
+      return { nxProvider: this.nxProvider, name: this.searchName };
+    }
+
+    _formChanged(e) {
+      this.results = e.detail.value && this._grabResults([e.detail.value]);
+    }
+
+    _grabResults(els) {
+      if (!Array.isArray(els) || els.length === 0) {
+        return;
+      }
+      // let's find the results element on the local and light dom of the current elements
+      let i;
+      let el;
+      for (i = 0; i < els.length; i++) {
+        el = els[i];
+        let results = dom(el).querySelector('nuxeo-results');
```

Comment:
Did we end up creating a ticket to clean this up afterwards?

Reply:
No, just because we ended up going with an approach that didn't require changes on the existing mechanism. I can create it anyway. The goal of that would be to completely review the way we look for the results in the loaded layout, right? Anything else?

Reply:
The goal would be to review the way nuxeo-search-results-layout picks the nuxeo-results element. Ideally, it would check for a given attribute and value (such as an given id, for example).

Reply:
ELEMENTS-1321: Review layout loading mechanism on nuxeo-search-results-layout

Reply:
thank you 🙏

Reply:
@nelsonsilva: We want the `nuxeo-results` (or some other element exposing the same API) inside that `element`.